File: fix_gsc.js
const fs = require('fs');
let c = fs.readFileSync('script_v5.js', 'utf8');

c = c.replace(/wstSetGscBadge\('syncing', '\.\.\.'\);(\r?\n)+(\s*)try \{/g, `wstSetGscBadge('syncing', '...');\n\n$2// Doi Firebase load xong\n$2let waitCount = 0;\n$2while (!window._fbDataLoaded && waitCount < 30) {\n$2  await new Promise(r => setTimeout(r, 500));\n$2  waitCount++;\n$2}\n\n$2try {`);

c = c.replace(/wstSetGscBadge\('nomatch'\);(\r?\n)+(\s*)return;/g, `wstSetGscBadge('nomatch');\n$2localStorage.setItem('gsc_last_global_sync_date', todayVN());\n$2return;`);

fs.writeFileSync('script_v5.js', c);
console.log('Fixed script_v5.js');

File: fix_gsc.py
import re

with open('script_v5.js', 'r', encoding='utf-8') as f:
    content = f.read()

# 1. Wait for Firebase data before syncing
target1 = r"(wstSetGscBadge\('syncing',\s*'\.\.\.'\);(?:[\r\n]+)(?:\s*)try\s*\{)"
replacement1 = r"""wstSetGscBadge('syncing', '...');

  let waitCount = 0;
  while (!window._fbDataLoaded && waitCount < 30) {
    await new Promise(r => setTimeout(r, 500));
    waitCount++;
  }

  try {"""
content = re.sub(target1, replacement1, content)


# 2. Mark gsc_last_global_sync_date if no sites on GSC
target2 = r"(wstSetGscBadge\('nomatch'\);(?:[\r\n]+)(?:\s*)return;)"
replacement2 = r"""wstSetGscBadge('nomatch');
      localStorage.setItem('gsc_last_global_sync_date', todayVN());
      return;"""
content = re.sub(target2, replacement2, content)

with open('script_v5.js', 'w', encoding='utf-8', newline='') as f:
    f.write(content)

print("Done")
